metrics/network/network-metrics-memory-pss.sh

```diff
@@ -47,32 +46,42 @@ function pss_memory {
 	# have server and client running for sometime and we
 	# need to avoid to measure at the beginning of the running
 	local middle_time=5
-	# Name of the containers
-	local server_name="network-server"
-	local client_name="network-client"
 	# Arguments to run the client
 	local extra_args="-d"
 
-	setup
+	# Initialize/clean environment
+	init_env
+
 	local server_command="iperf -p ${port} -s"
-	local server_address=$(start_server "$server_name" "$image" "$server_command")
+	local server_address=$(start_server "$image" "$server_command")
 
 	local client_command="iperf -c ${server_address} -t ${total_time}"
-	start_client "$extra_args" "$client_name" "$image" "$client_command" > /dev/null
+	start_client "$image" "$client_command" "$extra_args"
 
 	# Measurement after client and server are more stable
 	echo >&2 "WARNING: sleeping for $middle_time seconds in order to have server and client stable"
 	sleep ${middle_time}
+
+	# Check the runtime in order to determine which process will
+        # be measured about PSS memory consumption.
+        if [ "$RUNTIME" == "runc" ]; then
+                process="iperf"
+        elif [ "$RUNTIME" == "cor" ] || [ "$RUNTIME" == "cc-runtime" ]; then
+                process="$QEMU_PATH"
+        else
+                die "Unknown runtime: $RUNTIME"
+        fi
+
 	local memory_command="smem --no-header -c pss"
-	${memory_command} -P ${QEMU_PATH} > "$result"
```

Comment:
ah - subtle bug you found/fixed?, missing the '^', so maybe this counted the smem tool footprint as well as the runtime footprint!?

Reply:
yeah, and in the case of runc execution, it returns that value :(, then even if there is no any qemu instance running it returned a result :(